tests/GraphNodes/GraphEventTest.php

```diff
@@ -0,0 +1,59 @@
+<?php
+/**
+ * Copyright 2014 Facebook, Inc.
+ *
+ * You are hereby granted a non-exclusive, worldwide, royalty-free license to
+ * use, copy, modify, and distribute this software in source code or binary
+ * form for use in connection with the web services and APIs provided by
+ * Facebook.
+ *
+ * As with any software that integrates with the Facebook platform, your use
+ * of this software is subject to the Facebook Developer Principles and
+ * Policies [http://developers.facebook.com/policy/]. This copyright notice
+ * shall be included in all copies or substantial portions of the software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ */
+namespace Facebook\Tests\GraphNodes;
+
+use Facebook\FacebookResponse;
+use Mockery as m;
+use Facebook\GraphNodes\GraphObjectFactory;
+
+class GraphEventTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @var FacebookResponse
+     */
+    protected $responseMock;
+
+    public function setUp()
+    {
+        $this->responseMock = m::mock('\\Facebook\\FacebookResponse');
+    }
+
+    public function testDatesGetCastToDateTime()
+    {
+        $dataFromGraph = [
+            'start_time' => '2015-04-14T11:00:00+0300',
+        ];
+
+        $this->responseMock
+            ->shouldReceive('getDecodedBody')
+            ->once()
+            ->andReturn($dataFromGraph);
+        $factory = new GraphObjectFactory($this->responseMock);
+        $graphObject = $factory->makeGraphEvent();
+
+        $start_time = $graphObject->getStartTime();
+
+        $this->assertInstanceOf('DateTime', $start_time);
+    }
```

Comment:
You should add some tests that check that all the fields that return a subtype are being casts as the proper subtypes. [See the page tests](https://github.com/facebook/facebook-php-sdk-v4/blob/master/tests/GraphNodes/GraphPageTest.php#L66) for an example. Although I just noticed that that test is using `\\`'s, but it should just be `\`. (Somebody want to fix that? #lowhangingfruit) :)